### ISSUE TITLE
Force one role per user in OpenSALT?

### ISSUE THREAD

**Issue by brandonopened** [opensalt/opensalt#384]
As a user of any type in OpenSALT, THEN I will only be able to be one role. 

Detail: Currently we allow users to be more than one role. I do not know the specifics of the interaction problems this may create, but as we create more administrative functions, I feel roles should be more specifically given to people. Plus, most usage rights simply escalate upward (eg an Admin has more rights than an editor, a Super Editor has more rights than an Admin etc)

**Comment by roverwolf** [comment]
Another idea has come up on this topic.  Allow Super Users to create their own roles based on the privileges available.

For the scenario above, the Super User could create a "User Administration" role for the secretary that would allow be granted the ability to manage users but not the ability to edit frameworks.

This would allow for the flexability to have different privilege sets than the defaults while at the same time making it so that only one role should be needed per user.

**Comment by jmarks** [comment]
I agree with David. In fact, it seems that in the long term a user might need both belong to multiple organization and have different roles in those organizations. For example:

As a registered user (non visitor) I want to be an editor for one organization and a reviewer or admin for another.

**Comment by brandonopened** [comment]
I agree in theory... so we'd need to create a page that allows custom permissions within a user type... and the current user types would basically just be pre-checked off sets of permissions. 
Essentially build out this chart of user actions 

![user management](https://user-images.githubusercontent.com/14943151/36746262-c2bfa854-1be9-11e8-9853-a3377074e893.PNG)

**Comment by roverwolf** [comment]
Yes.  I am thinking along the lines of something akin to Drupal's permissioning where the current set would be a default set of roles provided and then could be customized or additional roles added by a Super User.
